packages/extension/src/background/index.ts

```diff
@@ -15,6 +27,95 @@ const cacheAmplitudeDeviceId = async ({
   }
 };
 
+const excludedCompanionOrigins = [
+  'https://twitter.com',
+  'https://www.google.com',
+  'chrome-extension://',
+];
+
+const isExcluded = (origin: string) =>
+  excludedCompanionOrigins.some((e) => origin.includes(e));
+
+const sendBootData = async (req, sender) => {
+  if (isExcluded(sender?.origin)) {
+    return;
+  }
+  const cacheData = JSON.parse(localStorage.getItem('boot:local'));
+  const { data, settings } = await getBootData('companion', sender?.tab?.url);
+  let settingsOutput = settings;
+  if (!cacheData?.user || !('providers' in cacheData?.user)) {
+    settingsOutput = { ...settingsOutput, ...cacheData?.settings };
+  }
+  await browser.tabs.sendMessage(sender?.tab?.id, {
+    data,
+    settings: settingsOutput,
+  });
+};
+
+function handleMessages(message, sender) {
+  if (message.type === 'CONTENT_LOADED') {
+    sendBootData(message, sender);
```

Comment:
If the user has opted out we should not send a request to our server at all, I think. What do you think?

Reply:
The main issue is that this info only comes from the boot query.
We could do it based on the local cache, but it might not always be the latest value.

Reply:
I'm ok with this tradeoff. If the user opted out we should reduce the overhead of this extension to minimum